recipes/xqilla/all/conanfile.py

```diff
@@ -56,10 +56,11 @@ def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("xerces-c/3.2.3")
+        # xerces-c is public, see https://c3i.jfrog.io/artifactory/misc-v2/logs/pr/17332/3-linux-gcc/xqilla/2.3.4/ed5ac4a4d7ee70de0440268155f960162d458f1d-test.txt
+        self.requires("xerces-c/3.2.4", transitive_headers=True, transitive_libs=True)
```

Comment:
``transitive_libs=True`` should be exceptional and require a conceptual explanation besides the CI link. Not defining ``transitive_libs=True`` doesn't mean that the library is not propagated in normal scenarios like app->static->static. 
The error seen there in CI when ``shared``, might be fixed in ``test_package/CMakeLists.txt`` with ``target_link_libraries(... PUBLIC ...)``

Reply:
xerces-c is public (headers and symbols in binary), and I don't have the time to find exactly which symbol of xerces-c exactly and where in public headers of xqilla it comes from, but the error is a clear indication that headers and libs are public.

> The error seen there in CI when shared, might be fixed in test_package/CMakeLists.txt with target_link_libraries(... PUBLIC ...)

It would be a lie for the sole purpose to pass v2 pipeline and avoid a required transitive libs that would break soon or later for consumers anyway. It's obvious looking at test_package.cpp that test package itself doesn't directly depend on xerces-c symbols.

Reply:
@memsharded  I don't see in [documentation](https://docs.conan.io/2.0/reference/conanfile/methods/requirements.html#transitive-libs) anything about `transitive_libs=True` should be exceptional. can you point me to the source?